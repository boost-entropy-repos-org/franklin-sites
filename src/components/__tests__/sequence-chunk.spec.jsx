import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SequenceChunk from '../sequence-chunk';

configure({ adapter: new Adapter() });

describe('Sequence chunk component', () => {
  test('should render without highlights', () => {
    const textSize = { width: 10, height: 10 };
    const wrapper = shallow(
      <SequenceChunk sequence="ABCDEFGHIJ" textSize={textSize} chunkSize={10} chunkNumber={2} />,
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render with highlights', () => {
    const wrapper = shallow(
      <SequenceChunk
        sequence="ABCDEFGHIJ"
        textSize={{ width: 10, height: 10 }}
        highlights={[
          {
            aminoAcids: ['A', 'F', 'I'],
            colour: 'blue',
          },
        ]}
        chunkSize={10}
        chunkNumber={2}
      />,
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
