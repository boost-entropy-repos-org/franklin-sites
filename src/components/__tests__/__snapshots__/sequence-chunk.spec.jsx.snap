// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sequence chunk component should render with highlights 1`] = `
"<Fragment>
  <svg height={30} width={100}>
    <text x={100} y={10} className=\\"sequence__sequence__chunk__counter\\">
      30
    </text>
    <text x=\\"0\\" y={20} className=\\"sequence__sequence__chunk__sequence\\">
      ABCDEFGHIJ
    </text>
    <rect x={0} y={22} width={10} height={2} fill=\\"blue\\" />
    <rect x={50} y={22} width={10} height={2} fill=\\"blue\\" />
    <rect x={80} y={22} width={10} height={2} fill=\\"blue\\" />
  </svg>
</Fragment>"
`;

exports[`Sequence chunk component should render without highlights 1`] = `
"<Fragment>
  <svg height={30} width={100}>
    <text x={100} y={10} className=\\"sequence__sequence__chunk__counter\\">
      30
    </text>
    <text x=\\"0\\" y={20} className=\\"sequence__sequence__chunk__sequence\\">
      ABCDEFGHIJ
    </text>
  </svg>
</Fragment>"
`;
